UI Improvements

aruan20 opened the pull request:
Implemented/updated several UI features such as:
- Adding the logo to the header
- Updating the layout of the landing page and the search results page
- Ensuring the pagination only appears on the search results
- Moving the search bar to the header when displaying results
- Ensuring the pagination and footer stays at the bottom of the page

Landing page: 
![image](https://github.com/user-attachments/assets/eea7b34d-10c9-430d-9c93-13dec349eae1)

Search results: 
![Screenshot 2025-04-11 at 4 40 38 AM](https://github.com/user-attachments/assets/92cc0cc2-eb31-4dce-a697-fa8df97b6605)

aruan20 committed:
Add logo to header

Added logo to header and updated it to use the correct logo based on dark/light theme.

aruan20 committed:
Update header and home page

Updated home page with new layout and updated the header to include the search bar.

aruan20 committed:
Fix image sizing of logo

Fixed sizing by cropping image of logo and also adjusted the spacing between the logo and the search bar.

aruan20 committed:
Allow user to search when pressing enter

Updated search bar so that search is submitted when the user presses enter after typing

aruan20 committed:
Fix logo and remove pagination on landing page

Fixed error where logo does not use correct image when theme switches. Removed pagination from the landing page and set it to only appear when showing results.

aruan20 committed:
Make header and footer sticky

Made header, footer, and pagination sticky so that they stay at the top and bottom of the page.
Need to fix pagination - disappears and reappears when selecting a different page.

aruan20 committed:
Update search results page

Updated the search results page with new user interface.
Restructured code so that the pagination does not show when loading.
Fixed styling so that footer stays at the bottom of the page.

etbala reviewed (changes requested):
Looks Good. A couple things:
- For some reason when I load it on my machine I get the light mode logo on the dark mode background (my system pref is dark mode).
- Would you mind changing the `Made by RCOS` text to `An RCOS Project`
- Also for some reason I was only able to run `npm run dev`, as `npm run build` errored out. Not sure if this is important at all, just thought I'd mention.

aruan20 committed:
Fix logo and theme match

Fixed issue where the light logo is displayed when the system theme is set to dark.

aruan20 committed:
Fix search issue

Fixed issue where page reloads every time a letter was added/deleted from the search input.
Updated how logo was imported and used so that it was consistent across pages.
Updated overall styling.

etbala reviewed (approved):
LGTM
